nemo/utils/callbacks/torch_dist_async.py

```diff
@@ -0,0 +1,220 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
```

Comment:
@dimapihtar this whole file won't be necessary once the following MR is available in MCore: https://gitlab-master.nvidia.com/ADLR/megatron-lm/-/merge_requests/1380